### PR TITLE
Add utility function for reading and visualizing scene data in binary format.

### PR DESCRIPTION
**Pull request by johnwlambert**
Run:
```
python gtsfm/visualization/view_scene.py --output_dir tanks_temples-2023-08-03-colmap-results/Barn/sparse/0 --file_format bin
```
See:
Screen Shot 2023-08-03 at 11 55 24 PM

### CODE REVIEW

File: gtsfm/utils/io.py [hunk 1]
```diff
@@ -450,13 +457,40 @@ def read_points_txt(fpath: str) -> Tuple[Optional[np.ndarray], Optional[np.ndarr
...
+    Returns:
+        5-tuple of:
+            wTi_list: List of camera poses for each image.
+            img_fnames: List of image file names, for each image.
+            calibrations: Calibration object for each camera.
+            point_cloud: Float array of shape (N,3) representing per-point x/y/z coordinates.
+            rgb: Uint8 array of shape (N,3) representing per-point colors.
+    """
+    if file_format == "txt":
+        points_fpath = f"{data_dir}/points3D.txt"
```
Comment (akshay-krishnan) [inline gtsfm/utils/io.py:481]: maybe just check which format exists in the directory? do we need another parameter here?

Comment (johnwlambert) [reply gtsfm/utils/io.py:490]: Good suggestion, updated now to automatically infer file format of scene data based on file existence.

File: gtsfm/utils/io.py [hunk 2]
```diff
@@ -450,13 +457,49 @@ def read_points_txt(fpath: str) -> Tuple[Optional[np.ndarray], Optional[np.ndarr
...
+        images_fpath = f"{data_dir}/images.txt"
+        cameras_fpath = f"{data_dir}/cameras.txt"
+        wTi_list, img_fnames = read_images_txt(images_fpath)
+        calibrations = read_cameras_txt(cameras_fpath)
+        point_cloud, rgb = read_points_txt(points_fpath)
+
+    elif file_format == "bin":
+        cameras, images, points3d = colmap_io.read_model(path=data_dir, ext=".bin")
```
Comment (akshay-krishnan) [inline gtsfm/utils/io.py:498]: could we use this for .txt too and do the `colmap2gtsfm` ? if not, maybe add a comment/TODO why?

**Review by akshay-krishnan — APPROVED** [review]

File: gtsfm/utils/io.py (hunk 2 above)
Comment (johnwlambert) [reply gtsfm/utils/io.py:498]: Good point, I'll add a TODO in a follow-up.

File: gtsfm/utils/io.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/utils/io.py:490]: (Marking as resolved).

**Pull request merged**